### PR TITLE
Feature/llamacpp

### PR DESCRIPTION
**Pull request by ashish-aesthisia**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,5 +4,14 @@ models/.locks
...
 
+src/original_model/*
+src/quantized_model/*
+
+src/llama_cpp
+
+#ignore build folder(for cmake)
+build
+
+
```
Comment (ashish-aesthisia) [inline .gitignore:15]: Remove extra space

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Requirements file
+set(REQUIREMENTS_FILE "requirements.txt")
+
+# Llama directory
+set(LLAMA_DIR "../src/llama_cpp")
+
+# Check for Python and Git using CMake's FIND_PACKAGE
+find_package(PythonLibs REQUIRED)
+find_package(Git REQUIRED)
+
```
Comment (ashish-aesthisia) [inline CMakeLists.txt:19]: Make sure, these two commands are not required with cmake

touch $(LLAMA_DIR)/__init__.py
 cp $(LLAMA_DIR)/convert-hf-to-gguf.py $(LLAMA_DIR)/convert_hf_to_gguf.py

File: configs/config.ini [hunk 3]
```diff
@@ -1,4 +1,4 @@
 [Settings]
-execution_provider = 
-repo_id = 
+execution_provider = cpu
```
Comment (ashish-aesthisia) [inline configs/config.ini:2]: Remove values, initially we keep variables not the values. 
Revert this change

File: src/quantize.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import subprocess
+from huggingface_hub import snapshot_download
+# from src.llama_cpp import convert_hf_to_gguf
```
Comment (ashish-aesthisia) [inline src/quantize.py:3]: Remove commented code & format file for extra line breaks & spaces

File: webui.py [hunk 5]
```diff
@@ -125,55 +117,58 @@ def parse_args():
                 chatbot = gr.Chatbot(elem_id="chatbot-container")
                 msg = gr.Textbox(label="Prompt")
                 stop = gr.Button("Stop")
-    llm_chain, llm = init_chain(model, tokenizer)
+
+    llm_chain, llm = init_llm_chain("./src/quantized_model\stabilityai__stable-code-instruct-3b.gguf")
```
Comment (ashish-aesthisia) [inline webui.py:121]: this path should be dynamic, based on what repo id user has added, return converted models path from quantize.py

**Review by Subhanshu0027 — APPROVED** [review]
LGTM!
